pulsar-broker/src/main/java/org/apache/pulsar/broker/service/Consumer.java

```diff
@@ -448,8 +448,8 @@ public Future<Void> sendMessages(final List<? extends Entry> entries,
     }
 
     private void incrementUnackedMessages(int unackedMessages) {
-        if (Subscription.isIndividualAckMode(subType)
-                && addAndGetUnAckedMsgs(this, unackedMessages) >= getMaxUnackedMessages()
+        if (addAndGetUnAckedMsgs(this, unackedMessages) >= getMaxUnackedMessages()
```

Comment:
Making state mutations outside of if conditions makes it easier to read and maintain the code in the future.
```suggestion
        int updatedUnackedMessages = addAndGetUnAckedMsgs(this, unackedMessages);
        if (updatedUnackedMessages >= getMaxUnackedMessages()
```